src/olympia/devhub/views.py

```diff
@@ -1439,7 +1439,8 @@ def _submit_upload(request, addon, channel, next_listed, next_unlisted,
             addon.update(**addon_update)
         # auto-sign versions (the method checks eligibility)
         auto_sign_version(version, is_beta=is_beta)
-        next_url = (next_listed if channel == amo.RELEASE_CHANNEL_LISTED
+        next_url = (next_listed
+                    if channel == amo.RELEASE_CHANNEL_LISTED and not is_beta
                     else next_unlisted)
```

Comment:
I don't have any good suggestions so I'm not blocking the review on it, but from the name you'd assume it's safe for `next_unlisted` to be an unlisted-specific URL, but with this change it's not, since we could be dealing with a beta... Maybe add a comment somewhere ? Maybe in the docstring ?